src/shogun/machine/gp/EPInferenceMethod.cpp

```diff
@@ -465,21 +465,10 @@ SGVector<float64_t> CEPInferenceMethod::get_derivative_wrt_kernel(
 	// create eigen representation of the matrix Q
 	Map<MatrixXd> eigen_F(m_F.matrix, m_F.num_rows, m_F.num_cols);
 
+	REQUIRE(param, "Param not set\n");
 	SGVector<float64_t> result;
-
-	if (param->m_datatype.m_ctype==CT_VECTOR ||
```

Comment:
so you remove those checks everywhere.
Is the idea that if param is set, then all those are redundant? Or are these things checked elsewhere anyways?